sanic/signals.py

```diff
@@ -131,3 +132,43 @@ def finalize(self, do_compile: bool = True):
             signal.ctx.event = asyncio.Event()
 
         return super().finalize(do_compile=do_compile)
+
+    def _build_event_parts(self, event: str) -> Tuple[str, str, str]:
+        parts = path_to_parts(event, self.delimiter)
+        if (
+            len(parts) != 3
+            or parts[0].startswith("<")
+            or parts[1].startswith("<")
+        ):
+            raise InvalidSignal("Invalid signal event: %s" % event)
+
+        current = Reservation(*parts)
+        for reservation in RESERVED_EVENTS:
+            if reservation.namespace == "*":
```

Comment:
None of the Reservations in `RESERVED_EVENTS` have namespace `*`. Why do we check that here?

Should it be `if current.namespace == "*"`? If so, should that go outside the for-loop?

Reply:
Or is the intention that if we put `*.foo.bar` in `RESERVED_EVENTS` then all added events are invalid?

Reply:
Yes, the idea would be that `*.foo.bar` would kill the ability to have any custom namespace.